fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/typing/Type.java

```diff
@@ -1475,20 +1532,17 @@ public void addProtoField(@Nonnull final TypeRepository.Builder typeRepositoryBu
                                   @Nonnull final Optional<String> typeNameOptional,
                                   @Nonnull final FieldDescriptorProto.Label label) {
             Objects.requireNonNull(elementType);
-            //
-            // If the inner type is nullable, we need to create a nested helper message to keep track of
-            // nulls.
-            //
-            if (definesNestedProto()) {
```

Comment:
I guess I removed `definesNestedProto` part because we said we don't need to support arrays with nulls in it. Is there any backward compatibility issue?

Reply:
I don't think so. This feature comes from the Norse prototype but nothing really depends on that particular feature being there. We can remove that.